packages/ui/src/table/TableBadge.tsx

```diff
@@ -1,19 +1,53 @@
-/* eslint-disable unused-imports/no-unused-imports */
+import React, {forwardRef, useImperativeHandle, useState} from "react";
+import {View} from "react-native";
 
-import React, {FC} from "react";
-import {Text, View} from "react-native";
+import {Badge} from "../Badge";
+import {TableBadgeProps} from "../Common";
+import {SelectField} from "../SelectField";
 
-import {Box} from "../Box";
-
-export interface TableBadgeProps {
-  value: string;
-  isEditing?: boolean;
+export interface TableBadgeHandles {
+  handleSave: () => void | Promise<void>;
 }
 
-export const TableBadge: FC<TableBadgeProps> = ({value, isEditing}) => {
-  return (
-    <View>
-      <Text>TableBadge</Text>
-    </View>
-  );
-};
+// TODO: Support error state in TableBadge
+export const TableBadge = forwardRef<TableBadgeHandles, TableBadgeProps>(
+  (
+    {value, badgeStatus = "info", badgeIconName, isEditing = false, editingOptions, onSave},
+    ref
+  ) => {
+    const [selected, setSelected] = useState<string | undefined>(value);
+
+    useImperativeHandle(ref, () => ({
```

Comment:
I'm learning a new hook here. Why do we need to use the imperative handle here if there's no underlying DOM node that we're ref-ing?

Reply:
I'm not entirely sure, I think this was going to be related to saving. I can remove it.